Program.cs

```diff
@@ -15,252 +15,273 @@ public static void printsContentsRefrigerator(List<Refrigerator> allfrige)
-                Console.WriteLine("2. " + (Item.Kashrot)1);
-                Console.WriteLine("3. " + (Item.Kashrot)2);
-                string input = Console.ReadLine();
-                if (int.TryParse(input, out kashrot) && kashrot > 0 && kashrot < 4)
-                {
-                    return (Item.Kashrot)(--kashrot);
-
-                }
-                else { Console.WriteLine("Please,enter valid input"); }
+                throw new Exception("Invalid input");
             }
+
         }
-        public static Item.TypeItem checkTypeItemInput()
+        public static int GetKosherInput()
+        {
+            int kashrot;
+            string userInput;
+            Console.WriteLine("Pease select the Kashrot of the item:");
+            Console.WriteLine("1. " + (Item.Kashrot)0);
+            Console.WriteLine("2. " + (Item.Kashrot)1);
+            Console.WriteLine("3. " + (Item.Kashrot)2);
+            userInput = Console.ReadLine();
+            if (!(int.TryParse(userInput, out kashrot)) || kashrot < 1 || kashrot > 3)
+                return -1;
+            return --kashrot;
+
+        }
+        public static int GetTypeItemInput()
         {
             int typeItem;
-            while (true)
-            {
-                Console.WriteLine("lease select the Type of the item:");
-                Console.WriteLine("1. " + (Item.TypeItem)0);
-                Console.WriteLine("2. " + (Item.TypeItem)1);
-                string inp = Console.ReadLine();
-                if (int.TryParse(inp, out typeItem) && typeItem > 0 && typeItem < 3)
-                {
-                    return (Item.TypeItem)(--typeItem);
-                }
-                else { Console.WriteLine("Please,enter valid input"); }
-            }
+            string userInput;
+            Console.WriteLine("Pease select the Type of the item:");
+            Console.WriteLine("1. " + (Item.TypeItem)0);
+            Console.WriteLine("2. " + (Item.TypeItem)1);
+            userInput = Console.ReadLine();
+            if (!(int.TryParse(userInput, out typeItem)) || typeItem < 1 || typeItem > 2)
+                return -1;
+            return --typeItem;
         }
-        public static String checkNameItemInput()
+        public static String GetNameItemInput()
         {
             string nameInput = "";
             Console.WriteLine("Name of the Item");
             nameInput = Console.ReadLine();
             while (nameInput.Length <= 0 || nameInput.Length > 50)
-                {
-                    Console.WriteLine("Please,enter valid input");
-                    nameInput = Console.ReadLine();
-                }
+            {
+                Console.WriteLine("Please,enter valid input");
+                nameInput = Console.ReadLine();
+            }
             return nameInput;
 
-            
+
         }
-        public static DateTime checkExpiryDatedInput()
+        public static object GetExpiryDatedInput()
```

Comment:
avoid static method in such an exercise like this.
the main purpose of static is to exposure the method to all layers of the code.